pkg/controller/factory/builder.go

```diff
@@ -63,12 +63,13 @@ func BuildRSM(cluster *appsv1alpha1.Cluster, synthesizedComp *component.Synthesi
 	// build labels
 	labels := constant.GetKBWellKnownLabelsWithCompDef(compDefName, clusterName, compName)
```

Comment:
Consider in the 0.9 version of the API, should the AppNameLabelKey in GetKBWellKnownLabelsWithCompDef continue to use componentDefinition or clusterDefinition?